### PR TITLE
[Add] Categories tab UI

### PR DESCRIPTION
**Pull request by truyentd-2427**
## What's this PR do ?

- Implement Categories tab UI

## Checklist

- [ ] Write/Update Unit Test (if have)
- [ ] Fill link PR into ticket and the opposite
- [ ] Validate UI/Model/API with latest specs
- [ ] Add UI pics to this PR (if have).

## Demo

https://github.com/namnh-0652/flutter_movie_clean/assets/77828973/4481a34c-7535-48fc-b0fa-44dc3bcbdfec

### CODE REVIEW

File: lib/pages/categories/categories_page.dart [hunk 1]
```diff
@@ -1,4 +1,10 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_movie_clean/gen/colors.gen.dart';
+import 'package:flutter_movie_clean/pages/categories/components/filter_chip_group.dart';
+import 'package:flutter_movie_clean/shared/extensions/context_ext.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+import 'components/segmented_control.dart';
```
Comment (namnh-0652) [inline lib/pages/categories/categories_page.dart]: `import 'package:flutter_movie_clean/...'`

File: lib/pages/categories/categories_page.dart [hunk 2]
```diff
@@ -9,15 +15,69 @@ class CategoriesPage extends StatefulWidget {
...
   Widget build(BuildContext context) {
     super.build(context);
-    return const Center(
-      child: Text("CategoriesPage"),
+    return Scaffold(
+      backgroundColor: AppColors.black,
+      body: Column(
+        mainAxisSize: MainAxisSize.max,
+        children: [
+          const SizedBox(height: 42),
```
Comment (namnh-0652) [inline lib/pages/categories/categories_page.dart]: should we use .h, .w to make UI consistent?

File: lib/pages/categories/components/segmented_control.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        mainAxisSize: MainAxisSize.max,
+        children: widget.labels.mapIndexed((index, label) {
+          return _buildSegmentItem(index, label);
+        }).toList(),
+      ),
+    );
+  }
+
+  Widget _buildSegmentItem(int segmentIndex, String label) {
+    return Expanded(
```
Comment (namnh-0652) [inline lib/pages/categories/components/segmented_control.dart:41]: Is it necessary to use Expanded here

Comment (truyentd-2427) [reply lib/pages/categories/components/segmented_control.dart:41]: @namnh-0652 
yes, according to the design below, we need to fill full the space of each segment item
Screen Shot 2023-08-29 at 13 13 55

**Review by namnh-0652 — APPROVED** [review]

**Pull request merged**